File: wrappers/java/src/main/java/org/hyperledger/indy/sdk/signus/SignusResults.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyJava;

/**
 * signus.rs results
 */
/**
 * Result classes for Signus operations.
 */
public final class SignusResults {

	private SignusResults() {

	}

	/**
	 * Result from calling createAndStoreMyDid.
	 */
	public static class CreateAndStoreMyDidResult extends IndyJava.Result {

		private String did, verkey, pk;
		CreateAndStoreMyDidResult(String did, String verkey, String pk) { this.did = did; this.verkey = verkey; this.pk = pk; }
		
		/**
		 * Gets the DID.
		 * 
		 * @return The DID.
		 */
		public String getDid() { return this.did; }
		
		/**
		 * Gets the verification key.
		 * 
		 * @return The verification key.
		 */
		public String getVerkey() { return this.verkey; }
		
		/**
		 * Gets the public key.
		 * 
		 * @return The public key.
		 */
		public String getPk() { return this.pk; }
	}

	/**
	 * Result from calling replaceKeys.
	 */
	public static class ReplaceKeysStartResult extends IndyJava.Result {

		private String verkey, pk;
		ReplaceKeysStartResult(String verkey, String pk) { this.verkey = verkey; this.pk = pk; }
		
		/**
		 * Gets the verification key.
		 * 
		 * @return The verification key.
		 */
		public String getVerkey() { return this.verkey; }
		
		/**
		 * Gets the public key.
		 * 
		 * @return The public key.
		 */
		public String getPk() { return this.pk; }
	}

	/**
	 * Result from calling encrypt.
	 */
	public static class EncryptResult extends IndyJava.Result {

		private byte[] encryptedMessage, nonce;
		EncryptResult(byte[] encryptedMessage, byte[] nonce) { this.encryptedMessage = encryptedMessage; this.nonce = nonce; }

		/**
		 * Gets the encrypted message.
		 *
		 * @return The encrypted message.
		 */
		public byte[] getEncryptedMessage() { return this.encryptedMessage; }

		/**
		 * Gets the nonce.
		 *
		 * @return The nonce.
		 */
		public byte[] getNonce() { return this.nonce; }
	}
}

File: wrappers/dotnet/indy-sdk-dotnet/Wrapper/SignusResults.cs
namespace Indy.Sdk.Dotnet.Wrapper
{
    /// <summary>
    /// Result of creating and storing my DID.
    /// </summary>
    public class CreateAndStoreMyDidResult
    {
        /// <summary>
        /// Initalizes a new CreateAndStoreMyDidResult.
        /// </summary>
        /// <param name="did">The DID created.</param>
        /// <param name="verKey">The verification key for the signature.</param>
        /// <param name="pk">The primary key for decrpytion</param>
        internal CreateAndStoreMyDidResult(string did, string verKey, string pk)
        {
            Did = did;
            VerKey = verKey;
            Pk = pk;
        }

        /// <summary>
        /// Gets the DID.
        /// </summary>
        public string Did { get; }

        /// <summary>
        /// Gets the verification key.
        /// </summary>
        public string VerKey { get; }

        /// <summary>
        /// Gets the primary key.
        /// </summary>
        public string Pk { get; }

    }

    /// <summary>
    /// Result of replacing keys.
    /// </summary>
    public class ReplaceKeysResult
    {
        /// <summary>
        /// Initializes a new ReplaceKeysResult.
        /// </summary>
        /// <param name="verKey">The verification key.</param>
        /// <param name="pk">The primary key.</param>
        internal ReplaceKeysResult(string verKey, string pk)
        {
            VerKey = verKey;
            Pk = pk;
        }

        /// <summary>
        /// Gets the verification key.
        /// </summary>
        public string VerKey { get; }

        /// <summary>
        /// Gets the primary key.
        /// </summary>
        public string Pk { get; }

    }

    /// <summary>
    /// The result of encryption.
    /// </summary>
    public class EncryptResult
    {
        /// <summary>
        /// Initializes a new EncryptionResult.
        /// </summary>
        /// <param name="encryptedMsg">The encrypted message.</param>
        /// <param name="nonce">The nonce.</param>
        internal EncryptResult(string encryptedMsg, string nonce)
        {
            EncryptedMsg = encryptedMsg;
            Nonce = nonce;
        }

        /// <summary>
        /// Gets the encrypted message.
        /// </summary>
        public string EncryptedMsg { get; }

        /// <summary>
        /// Gets the nonce.
        /// </summary>
        public string Nonce { get; }

    }
}
